src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs

```diff
@@ -1142,5 +1139,30 @@ public void ConsensusAndHeadersAreAtBlock5_Block6Presented_PartialValidationSucc
             peerIdsByTipHash[header6Hash].Should().Contain(ChainedHeaderTree.LocalPeerId);
             peerIdsByTipHash[header6Hash].Should().Contain(peerId);
         }
+
+        /// <summary>
+        /// Issue 29 @ Remove ChainHeader manually from a tree and call PartialValidationSucceeded,
+        /// make sure it returns nothing.
+        /// </summary>
+        [Fact]
+        public void ChainedHeaderIsRemovedFromTheTree_PartialValidationSucceededCalled_NothingIsReturned()
+        {
+            // Chain header tree setup. Initial chain has 5 headers.
+            // Example: h1=h2=h3=h4=h5.
+            const int initialChainSize = 5;
+            TestContext ctx = new TestContextBuilder()
+                .WithInitialChain(initialChainSize)
+                .Build();
+            ChainedHeaderTree cht = ctx.ChainedHeaderTree;
+            ChainedHeader chainTip = ctx.InitialChainTip;
+
+            // Manually remove chained header.
+            Dictionary<uint256, ChainedHeader> chainedHeaders = cht.GetChainedHeadersByHash();
+            chainedHeaders.Remove(chainTip.HashBlock);
+
+            // PartialValidationSucceeded should return null.
+            List<ChainedHeader> headersToValidate = cht.PartialValidationSucceeded(chainTip, out bool reorgRequired);
+            headersToValidate.Should().BeNull();
```

Comment:
I don't really agree with Ivan here. I think this test should be performed differently. I think we should have peer presents at least two headers. Those headers will be connected the tree. Then we save the first such connected block to variable X.  Then we simulate block downloaded for BOTH blocks and then we call disconnect on the peer, which will remove its chain from the tree. Then we call partial validation succeeded on X. 

@noescape00 I don't think the test otherwise makes sense.